CONTRIBUTING.md

```diff
@@ -137,9 +144,59 @@ The [packages/](packages/) directory contains the following public/published pac
+
+* [mockery](https://www.npmjs.com/package/mockery): Mockery is a simple module for mocking Node.js modules during testing. It allows you to replace real modules with mocks or stubs.
+  
+* [mocha](https://mochajs.org/): Mocha is a feature-rich JavaScript test framework that runs on Node.js and in the browser. It facilitates asynchronous testing, making it easy to write simple and flexible tests.
+
+* [chai](https://www.npmjs.com/package/chai): Chai is a BDD/TDD assertion library for Node.js and the browser. It can be paired with any JavaScript testing framework. We use it to create readable and expressive assertions.
+
+* [sinon](https://sinonjs.org/releases/v18/): Sinon is a standalone test spies, stubs, and mocks for JavaScript. It works with any test framework and integrates well with Mocha and Chai. We utilize Sinon for checking how functions are called during testing.
+
+* [sinon-chai](https://www.npmjs.com/package/sinon-chai): This library provides a set of custom assertions for using Sinon with Chai. It allows you to write more readable assertions for Sinon spies, stubs, and mocks.
+
+* [nyc](https://www.npmjs.com/package/nyc): NYC is a command-line utility for generating code coverage reports. It is often used with Mocha to ensure that tests cover as much code as possible.
+
+#### Mandatory Tools in the Testing Pipeline
+
+In addition to the testing frameworks, the following tools are mandatory for all testing pipelines:
+
+* [eslint](https://eslint.org/): ESLint is a static code analysis tool for identifying problematic patterns in JavaScript code. It is essential for maintaining code quality.
+* [license-check-and-add](https://www.npmjs.com/package/license-check-and-add): This tool ensures that all files in the codebase contain the required license headers. It is mandatory for all code submissions.
+
+#### Points to Note for Adding a Conforming Unit Test
+
+When writing unit tests, the following structure and practices are mandatory:
+
+1. **License Header**: All test files must include the project's license header.
+2. **'use strict' Directive**: Ensure strict mode is enabled in all test files.
+3. **Test Organization**:
+    * Use `describe` blocks to group related test cases.
+    * Use `it` statements for individual test cases.
+    * Nested `describe` blocks are encouraged for organizing complex test scenarios.
+4. **Consistent Test Naming**: Test descriptions should flow naturally, making it clear what behavior is being tested (e.g., 'should return the correct value when input is valid').
+5. **Mocking Guidance**: Be cautious with mocks that persist across tests. Always clean up after each test to avoid unexpected behavior.
+6. **Test Patterns**: Refer to the Fabric Unit tests for examples of recommended patterns and best practices.
+7. **Final Checks**: Always run all unit tests before submitting a PR and ensure no `.only` is left in the code, which would skip other tests.
+
+#### Integration Level Testing
```

Comment:
This should be 3 hashes, not 4 to ensure it is at the right level currently it comes under unit testing which is not where we would want it.

Reply:
Thanks for the catch, Done